### ISSUE TITLE
published R notebooks have broken output

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/cocalc#5764]
not sure what's going on, but even simple vectors or associative dictionaries are broken. seems like there is latex ouptut, which can't be rendered. It also looks a bit funny in the notebook itself, but at least it does render.

example:

```
y <- c(1,2,4)
y
```

```
df <- list(y = y)
df
```

ends up as 

![Screenshot from 2022-02-27 15-47-51](https://user-images.githubusercontent.com/207405/155887618-b6e5aea6-6c52-4c1b-9d27-5e24a7c5700d.png)

**Comment by williamstein** [comment]
This is weird.  The error says "No such environment "enumerate*".

There is indeed no such LaTeX environment.  Look at LaTeX itself with this input
```tex
...


\begin{enumerate*}
...
\end{enumerate*}
...
```

Output:
```
! LaTeX Error: Environment enumerate* undefined.

See the LaTeX manual or LaTeX Companion for explanation.
```

Also, `enumerate*` is such a weird idea. The entire point of "enumerate" is to make a numbered list (which makes no sense in math mode anyways).  The number part "numerate" is right in the name.  However, in latex a * in an env name turns of numbers. 

Mathjax also doesn't render enumerate or enumerate* environments, as far as I can tell. https://www.mathjax.org/#demo

Could this be a major bug in an update you made to the R kernel, or misconfiguration?

OK, I know what is happening.  R is creating output in 4 different formats as you can see here:

<img width="1637" alt="Screen Shot 2022-02-27 at 8 13 20 AM" src="https://user-images.githubusercontent.com/1276278/155890235-5407c846-f4ee-4974-a92f-93e2d49d05b5.png">

The share server is trying to render one version (which is not working), and the frontend notebook is rendering a different version (which is working).  I did recently impose a well-defined ordering on how rendering should work, but evidently there must be a mistake in that, or maybe it only applies to the share server.

Also, clearly the latex output from the R kernel is completely wrong.  That's definitely a bug in the R kernel.  

The fact that the R kernel outputs everything in all formats is annoying.  This reduces the performance and increases the file sizes by **an order of magnitude**.   Ugh.  I should change our backend to only transmit the best available format to the frontend (via sync) and ignore all the others.  Then there would be no harm for cocalc, and the advantage that maybe nbconvert can use the best format for what it is doing remains.

In this case, part of the problem is me assuming that the latex output format is math formulas, whereas the R kernel:

- assumes the latex output is not math mode.  As far as I can tell the nbformat docs just completely ignore this ambiguity: https://nbformat.readthedocs.io/en/latest/search.html?q=latex&check_keywords=yes&area=default
- and morever the actual latex it generates is invalid, since there is no enumerate* environment; or at least, maybe the R devs have some special style file, but I don't know.

Anyway, thanks for reporting this, and I think it'll be possible to fix, probably by just making latex get the lowest possible priority for rendering...

**Comment by williamstein** [comment]
Currently cocalc uses non-XSS-sanitized html in the jupyter notebook app to display this, and that makes a complicated style using an ordered list with no numbers to output a vector.  It's pretty abusive of html.   In the share server, we use the latex output since we lower the priority/value of html, due to XSS sanitation.  However, the latex is as mentioned above wrong.  I tried using markdown and html and due to sanitization html looks ridiculous; also, markdown is ridiculous, outputing an enumerated list:
```
1. 1
2. 2
3. 3
```

The only reasonable thing I can think of to do is prioritize text/plain at the highest priority... but only for the R kernels.   This is of course just ugly hacks to work around bugs.  However, it's really tricky in the code to do that too.
